### PR TITLE
The Pets Update

### PR DESCRIPTION
**Pull request by LikeMyBread**
This adds a lot of stuff, but mostly it adds pets. It'll require reprovisioning to both update models and add some pets to play with.

### CODE REVIEW

File: saylua/modules/pets/badwords.py [hunk 1]
```diff
@@ -1,2 +1,2 @@
...
...ead","knobed","knobend","knobhead","knobjocky","knobjokey","kock","kondum","kondums","kum","kummer","kumming","kums","kunilingus","labia","lmfao","lust","lusting","masterbate","masterbation","masterbations","masturbate","mofo","muff","mutha","muther","nazi","nigga","niggah","niggas","niggaz","nigger","niggers","nob","nobjokey","nobhead","nobjocky","nobjokey","numbnuts","nutsack","orgasim","orgasims","orgasm","orgasms","pecker","penis","phonesex","phuck","phuk","phuked","phuking","phukked","phukking","phuks","phuq","pimpis","piss","poop","porn","porno","pornography","pornos","prick","pricks","pron","pube","pusse","pussi","pussies","pussy","rape","rectum","retard","rimjaw","rimming","shit","sadist","schlong","screwing","scroat","scrote","scrotum","semen","sex","shag","shagger","shaggin","shagging","shemale","shit","shite","shited","shitey","shitfull","shithead","shiting","shitings","shits","shitted","shitter","shitters","shitting","shittings","shitty","skank","slut","sluts","smegma","smut","snatch","spac","spunk","testical","testicle","tits","titties","tosser","turd","twat","twathead","twatty","twunt","twunter","viagra","vulva","wank","whoar","whore","willies","willy","xrated","xxx"]
```
Comment (NoiSek) [inline saylua/modules/pets/badwords.py]: I can't believe I actually read all of these to see what the addition was.

File: saylua/modules/pets/views.py [hunk 2]
```diff
@@ -1,18 +1,83 @@
-from flask import render_template, redirect
+from flask import render_template, g, redirect, request, flash
+from saylua import db
+from .models.db import Pet
 
 
 def pet_profile(name):
```
Comment (NoiSek) [inline saylua/modules/pets/views.py]: There is way too much redundancy here, if you're performing the same action 3+ times why not just construct an output string and add / commit / flash all at once at the end?

I would like to see this at least partially simplified for somebody's sake further down the road

**Review by NoiSek — CHANGES_REQUESTED** [review]
Nice work! Functionally this is great stuff to see, made a note on one implementation detail I'm iffy on though.

File: saylua/modules/pets/views.py [hunk 3]
```diff
@@ -1,18 +1,83 @@
...
+    if request.method == 'POST':
+        if not g.user:
+            flash("You need to be logged in to do this!", "error")
+            return redirect("/login")
+        companion_id = request.form.get('accompany')
+        abandon_id = request.form.get('abandonee')
+        if companion_id:
+            companion = db.session.query(Pet).filter(Pet.id == companion_id).one_or_none()
+            if companion is None or companion.owner_id != g.user.id:
+                flash("You can't accompany this companion.")
```
Comment (tiffz) [inline saylua/modules/pets/views.py]: Psttt. Do flash('message', 'error') to make this case display as an error

File: saylua/modules/pets/views.py [hunk 4]
```diff
@@ -1,18 +1,83 @@
...
+from .models.db import Pet
 
 
 def pet_profile(name):
-    return render_template("profile.html")
+    if request.method == 'POST':
+        if not g.user:
+            flash("You need to be logged in to do this!", "error")
+            return redirect("/login")
+        companion_id = request.form.get('accompany')
```
Comment (tiffz) [inline saylua/modules/pets/views.py]: Personally, I feel like putting both abandoning and making pets active here is confusing and overloaded. I'd split these into two separate routes and make both redirect to the profile.

**Review by tiffz — CHANGES_REQUESTED** [review]
Yay!!!

**Review by NoiSek — APPROVED** [review]

File: saylua/modules/pets/views.py [hunk 5]
```diff
@@ -4,56 +4,58 @@
...
+    companion_id = request.form.get('accompany')
+    if companion_id:
+        companion = db.session.query(Pet).filter(Pet.id == companion_id).one_or_none()
+        if companion is None or companion.owner_id != g.user.id:
+            flash("You can't accompany this companion.")
+        else:
+            current_companion = g.user.companion
+            if not current_companion:
+                g.user.companion_id = companion.id
+                flash(companion.name + " is now accompanying you!")
```
Comment (NoiSek) [inline saylua/modules/pets/views.py]: Use dem dere formatted strings next time, `.format()` is your friend.

**Pull request merged**